programs/server/Server.cpp

```diff
@@ -2011,11 +2013,25 @@ void Server::updateServers(
     std::vector<ProtocolServerAdapter> & servers)
 {
     Poco::Logger * log = &logger();
-    /// Gracefully shutdown servers when their port is removed from config
+
     const auto listen_hosts = getListenHosts(config);
     const auto listen_try = getListenTry(config);
 
+    /// Remove servers once all their connections are closed
+    std::erase_if(servers, [&log](auto & server)
+    {
+        if (!server.isStopping())
+            return false;
+        size_t current_connections = server.currentConnections();
+        LOG_DEBUG(log, "Server {} (from one of previous reload): {} ({} connections)",
+            server.getDescription(),
+            !current_connections ? "finished" : "waiting",
+            current_connections);
+        return !current_connections;
+    });
```

Comment:
Looks like unnecessary code duplication. I mean you have the same code at the beginning and at the end of this function.

Reply:
Yes, but using separate `erase_if` servers that was already stopped during previous reload can be distinguished (and the message is different).

Reply:
Let's wrap the code in a separate function to do that without code duplication?

Reply:
There are only few lines of code, but OK, done.